src/krb5_asrep_fmt_plug.c

```diff
@@ -529,7 +529,8 @@ static int crypt_all(int *pcount, struct db_salt *salt)
 			}
 		}
 	}
-	new_keys = 0;
+	if (cur_salt->etype == 23)
+		new_keys = 0;
```

Comment:
I suggest instead move `new_keys = 0;` to line 448 or something

Reply:
Line 448 is `len = enc_to_utf16...` for the current code. I think you meant a different line number.

Reply:
I meant insert at line 448, moving the current line 448 down. It's pretty obvious if you look at the code there.